agimus_controller/agimus_controller/warm_start_base.py

```diff
@@ -1,24 +1,69 @@
 from abc import ABC, abstractmethod
+
 import numpy as np
+import numpy.typing as npt
 
 from agimus_controller.trajectory import TrajectoryPoint
 
 
 class WarmStartBase(ABC):
+    """
+    A template class for implementing methods that generate a warmstart for an optimal control problem.
+    The warmstart is expected to generate the initial values for state and control trajectories,
+    based on the initial robot state and a reference trajectory.
+
+    Attributes:
+        _previous_solution (list[TrajectoryPoint]): Stores the previous solution of the optimization problem.
```

Comment:
I am not 100% sure if we should document private attributes. Maybe an "inline" comment would be better?

Reply:
okay